Source/WebKit/ModelProcess/cocoa/ModelProcessModelPlayerProxy.mm

```diff
@@ -1017,6 +1046,67 @@ static void setIBLAssetOwnership(const String& attributionTaskID, REAssetRef ibl
     [m_modelRKEntity applyDefaultIBL];
 }
 
+#if ENABLE(MODEL_ELEMENT_IMMERSIVE)
+
+void ModelProcessModelPlayerProxy::ensureImmersivePresentation(CompletionHandler<void(std::optional<WebCore::LayerHostingContextIdentifier>)>&& completion)
+{
+    setImmersivePresentation(true);
+    ensureModelLoaded([weakThis = WeakPtr { *this }, completion = WTFMove(completion)] (bool loaded) mutable {
+        RefPtr protectedThis = weakThis.get();
+        if (!protectedThis)
+            return completion(std::nullopt);
+
+        if (loaded)
+            completion(protectedThis->layerHostingContextIdentifier().value());
+        else {
+            protectedThis->setImmersivePresentation(false);
+            completion(std::nullopt);
+        }
+    });
+}
+
+void ModelProcessModelPlayerProxy::exitImmersivePresentation(CompletionHandler<void()>&& completion)
+{
+    setImmersivePresentation(false);
+    completion();
+}
+
+void ModelProcessModelPlayerProxy::setImmersivePresentation(bool immersivePresentation)
+{
+    [CATransaction begin];
+    [CATransaction setDisableActions:YES];
```

Comment:
I think we might want to set the `WebActionDisablingCALayerDelegate` on this layer instead